### ISSUE TITLE
Consider using `spawn` and `channel` in specific async runtime.

### ISSUE THREAD

**Issue by Berrysoft** [crossterm-rs/crossterm#608]
I see that `EventStream` is using `std::thread` and `std::sync::mpsc` at now. It is of course an improvement according to #448. However, I'd like to suggest a possible improvement, with the usage of `spawn` and `channel`s in specific async runtime, for example, `tokio` or `async-std`. Therefore, the async runtime could manage the event stream thread.

The replacement in `tokio` is [`spawn`](https://docs.rs/tokio/1.12.0/tokio/fn.spawn.html) and [`mpsc`](https://docs.rs/tokio/1.12.0/tokio/sync/mpsc/index.html).

The replacement in `async-std` is [`spawn`](https://docs.rs/async-std/1.10.0/async_std/task/fn.spawn.html) and [`channel`](https://docs.rs/async-std/1.10.0/async_std/channel/index.html).

Both of them should be drop-in replacement, rewritting some functions to async ones. I'd like to open a pull request later, trying to add the implementation as an optional feature, and keep the original implementation as a fallback one.

**Comment by BoolPurist** [comment]
I agree that the underlying stream should not used to spawn a naked OS-thread outside any async runtime.

The (previous attempt to fix this)[https://github.com/crossterm-rs/crossterm/pull/609] seems to go back quite a while.

I am willing to pick this up from here. 

However, I have a few questions before starting the rewrite.

1. Is there now any interest in the rewrite ?
2. How to do it in terms of the compile features ? I have the following suggestion.
 - Only supporting tokio under the flag "event-stream"
 - The flag "event-stream" is working the context of tokio. 
   One new compiler flag "asyn-std" is then using asyn-std.
 - Two new flags "tokio" and "asyn-std" in which tokio and asyn-std are respectively used. 
   In this variant, I would deprecate the flag "event-stream" and type EventStream simply 
   since the implementation of this type with naked OS thread should not be used anymore 
   going forward in my opinion.

**Comment by Berrysoft** [comment]
> Is still there now any interest in the rewrite?

No, sorry. Now I think that it is easy enough for the users of tokio or other async runtimes to write another stream without spawing new threads.

**Comment by BoolPurist** [comment]
> > Is still there now any interest in the rewrite?
> 
> No, sorry. Now I think that it is easy enough for the users of tokio or other async runtimes to write another stream without spawing new threads.

Fair enough.  
One question remains for me:  shouldn't the usage of this feature then be discouraged/deprecated ?

**Comment by Berrysoft** [comment]
I think it may be better to add some comments in the doc about the spawnned thread, and suggest the user to write a runtime-specific one if they care about the overhead.

**Comment by BoolPurist** [comment]
I can go ahead and add this fact to the docs about the type EvenStream. While I am at it, I would also add those comments to the example with tokio or asyn-std using this type.
